src/test/test-hashmap-plain.c

```diff
@@ -49,6 +49,35 @@ TEST(hashmap_replace) {
         ASSERT_STREQ(r, "val5");
 }
 
+TEST(hashmap_ensure_replace) {
+        _cleanup_hashmap_free_ Hashmap *m = NULL;
+        _cleanup_free_ char *val1 = NULL, *val2 = NULL;
+        char *r;
```

Comment:
the name `r` is usually reserved for the local variable we store function return codes in, i.e. it's always defined as `int r` for us. Please use some other name, maybe `a` or `s` or whatever, just not `r`, it's just confusing otherwise.